### PR TITLE
changes for flow

### PR DESCRIPTION
**Pull request by Eldho-Abraham**

### CODE REVIEW

File: grid/src/index.js [hunk 1]
```diff
@@ -116,7 +116,11 @@ const Grid = (props) => {
     const [isLoaded, setIsLoaded] = useState(false);
 
     // Logic for searching in each column
-    const searchColumn = (column, original, searchText) => {
+    const searchColumn = (
+        column: Object,
+        original: Object,
+        searchText: Object
```
Comment (iCargopoc) [inline grid/src/index.js]: searchText is a string

File: grid/src/index.js [hunk 2]
```diff
@@ -191,7 +195,11 @@ const Grid = (props) => {
     };
 
     // Gets triggered when one row item is updated
-    const updateRowInGrid = (original, updatedRow, isSubComponentRow) => {
+    const updateRowInGrid = (
+        original: Object,
+        updatedRow: Object,
+        isSubComponentRow: Object
```
Comment (iCargopoc) [inline grid/src/index.js]: isSubComponentRow is boolean

File: grid/src/index.js [hunk 3]
```diff
@@ -219,7 +227,11 @@ const Grid = (props) => {
 
     // #region - Group sorting logic
     // Function to return sorting logic based on the user selected order of sort
-    const compareValues = (compareOrder, v1, v2) => {
+    const compareValues = (
+        compareOrder: Object,
```
Comment (iCargopoc) [inline grid/src/index.js]: compareOrder is string

File: grid/src/index.js [hunk 4]
```diff
@@ -249,9 +264,9 @@ const Grid = (props) => {
                 parentIdAttribute !== null &&
                 parentIdAttribute !== undefined
             ) {
-                let sortedTreeData = [];
+                let sortedTreeData = [null];
```
Comment (iCargopoc) [inline grid/src/index.js]: [null] should not be used

**Review by iCargopoc — CHANGES_REQUESTED** [review]

File: grid/src/index.js (hunk 1 above)
Comment (Eldho-Abraham) [reply grid/src/index.js]: done

File: grid/src/index.js (hunk 3 above)
Comment (Eldho-Abraham) [reply grid/src/index.js]: done

File: grid/src/index.js [hunk 5]
```diff
@@ -11,10 +11,10 @@ import Customgrid from "./Customgrid";
 // lazy styles inclusion via styleloader
 import __cmpStyles from "./Styles/main.scss";
 
-const processedData = (gridData, parentIdAttribute) => {
+const processedData = (gridData: Object, parentIdAttribute: Object): ?[] => {
```
Comment (iCargopoc) [inline grid/src/index.js:14]: gridData will be an array

File: grid/src/index.js [hunk 6]
```diff
@@ -214,7 +222,11 @@ const Grid = (props) => {
 
     // #region - Group sorting logic
     // Function to return sorting logic based on the user selected order of sort
-    const compareValues = (compareOrder, v1, v2) => {
+    const compareValues = (
+        compareOrder: string,
+        v1: Object,
```
Comment (iCargopoc) [inline grid/src/index.js:227]: v1 can be boolean/string/number etc...

File: grid/src/index.js [hunk 7]
```diff
@@ -214,7 +222,11 @@ const Grid = (props) => {
 
     // #region - Group sorting logic
     // Function to return sorting logic based on the user selected order of sort
-    const compareValues = (compareOrder, v1, v2) => {
+    const compareValues = (
+        compareOrder: string,
+        v1: Object,
+        v2: Object
```
Comment (iCargopoc) [inline grid/src/index.js:228]: v2 can be boolean/string/number etc...

File: grid/src/index.js [hunk 8]
```diff
@@ -232,7 +244,10 @@ const Grid = (props) => {
         return returnValue;
     };
     // Function to return sorted data
-    const getSortedData = (originalData, groupSortOptions) => {
+    const getSortedData = (
+        originalData: Object,
```
Comment (iCargopoc) [inline grid/src/index.js:248]: originalData will be array

File: grid/src/index.js [hunk 9]
```diff
@@ -232,7 +244,10 @@ const Grid = (props) => {
         return returnValue;
     };
     // Function to return sorted data
-    const getSortedData = (originalData, groupSortOptions) => {
+    const getSortedData = (
+        originalData: Object,
+        groupSortOptions: Object
```
Comment (iCargopoc) [inline grid/src/index.js:249]: groupSortOptions will be array

**Review by iCargopoc — CHANGES_REQUESTED** [review]

File: grid/src/index.js (hunk 6 above)
Comment (Eldho-Abraham) [reply grid/src/index.js:227]: object will be generic

File: grid/src/index.js (hunk 7 above)
Comment (Eldho-Abraham) [reply grid/src/index.js:228]: object will be generic

File: grid/src/index.js (hunk 8 above)
Comment (Eldho-Abraham) [reply grid/src/index.js:248]: done

File: grid/src/index.js (hunk 9 above)
Comment (Eldho-Abraham) [reply grid/src/index.js:249]: done

**Pull request closed without merging**